File: src/dto/movie.dto.ts
import { ApiProperty } from '@nestjs/swagger';

export class PostPutMovieDto {
  @ApiProperty()
  title: string;

  @ApiProperty({ enum: ['스릴러', '로맨스', '코믹', '액션'] })
  genre: string;

  @ApiProperty({ example: 'YYYY-MM-DD' })
  releaseDate: Date;

  @ApiProperty({ example: 'YYYY-MM-DD' })
  endDate: Date;

  @ApiProperty()
  isShowing: boolean;
}

File: src/main/kotlin/com/example/moviereviewapp/dto/MovieDTO.kt
package com.example.moviereviewapp.dto

import com.example.moviereviewapp.domain.Genre
import com.example.moviereviewapp.domain.Movie
import java.time.LocalDate

data class MovieDTO(
    val id: Long?,
    val title: String,
    val genre: Genre,
    val releaseDate: LocalDate,
    val endDate: LocalDate,
    val isShowing: Boolean,
    val registrationDate: LocalDate
) {
    fun toEntity(): Movie {
        return Movie(
            id = id,
            title = title,
            genre = genre,
            releaseDate = releaseDate,
            endDate = endDate,
            isShowing = isShowing,
            registrationDate = registrationDate
        )
    }
}